src/marqo/core/semi_structured_vespa_index/semi_structured_vespa_index.py

```diff
@@ -44,14 +44,20 @@ def to_marqo_document(self, vespa_document: Dict[str, Any], return_highlights: b
     def to_vespa_query(self, marqo_query: MarqoQuery) -> Dict[str, Any]:
         # Verify attributes to retrieve, if defined
         if marqo_query.attributes_to_retrieve is not None:
-            marqo_query.attributes_to_retrieve.append(common.VESPA_FIELD_ID)
+            # Always retrieve static fields content to extract non-string values from combined fields
```

Comment:
Will this cause search latency regression if they set `attributes_to_retrieve=[]`?

Reply:
Good point. I'll only add static fields when attributes_to_retrieve is not empty list. This will be a perf improvement comparing with the legacy unstructured where we always retrieve all fields.